cpp/src/IceSSL/OpenSSLEngine.cpp

```diff
@@ -451,20 +402,125 @@ OpenSSL::SSLEngine::initialize()
     }
     catch (...)
     {
-        //
-        // We free the SSL context regardless of whether the plugin created it
-        // or the application supplied it.
-        //
         SSL_CTX_free(_ctx);
         _ctx = nullptr;
         throw;
     }
 }
 
-SSL_CTX*
-OpenSSL::SSLEngine::context() const
+ClientAuthenticationOptions
+OpenSSL::SSLEngine::createClientAuthenticationOptions(const std::string&) const
 {
-    return _ctx;
+    return ClientAuthenticationOptions{
+        .clientSslContextSelectionCallback =
+            [this](const string&)
+        {
+            // Ensure the SSL context remains valid for the lifetime of the connection.
```

Comment:
Do we do decrement it?

Reply:
Yes in `TransceiverI::close` https://github.com/zeroc-ice/ice/blob/7c8290f686cc49e4a4a983e68aa4635a64fb5bb0/cpp/src/IceSSL/OpenSSLTransceiverI.cpp#L307-L310